### ISSUE TITLE
[Elixir] Implement new Concept Exercise: `dynamic-dispatch`

### ISSUE THREAD

**Issue by angelikatyborska** [exercism/v3#2803]
## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to replace `rpn-calculator-output` as the exercise that teaches dynamic dispatch (https://elixir-lang.org/getting-started/typespecs-and-behaviours.html#dynamic-dispatch). `rpn-calculator-output` should be modified to depend on `dynamic-dispatch` instead of teaching it. Make sure to reuse part of that exercise's introduction.

## Learning objectives

- Know that you module names are just atoms
- Know that you can call a function on a module even if its stored in a variable
- `apply`?

## Out of scope

?

## Concepts

- `dynamic-dispatch`

## Prerequisites

Depends on the story.

## Resources to refer to

https://elixir-lang.org/getting-started/typespecs-and-behaviours.html#dynamic-dispatch

## Story

Look through existing stories (`/reference/stories/`) to find inspiration or an existing story that can work for this exercise, or come up with a totally original idea.

## Implementing

Since dynamic dispatch depends on the existence of a few modules with the same function, we might want to provide that in the boilerplate instead of having the student implement those too.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

**Comment by angelikatyborska** [comment]
Does `apply` (https://hexdocs.pm/elixir/Kernel.html#apply/3) qualify as dynamic dispatch?

**Comment by tejasbubane** [comment]
> Does `apply` (https://hexdocs.pm/elixir/Kernel.html#apply/3) qualify as dynamic dispatch?

I think it does. If needed we can mention the fact that `apply` cannot be invoked with macros.

**Comment by angelikatyborska** [comment]
Migrated to the Elixir track repo: https://github.com/exercism/elixir/issues/558
